lib/snap/responses/search_response.ex

```diff
@@ -35,7 +35,17 @@ defmodule Snap.SearchResponse do
   end
 
   def build_aggregations(aggregations) when is_map(aggregations) do
-    Map.new(aggregations, fn {key, value} -> {key, Snap.Aggregation.new(value)} end)
+    Map.new(aggregations, fn
+      {key, %{"buckets" => _} = value} ->
+        {key, Snap.Aggregation.new(value)}
+
+      {key, %{"count" => _, "min" => _, "max" => _, "avg" => _, "sum" => _} = value} ->
+        {key, Snap.MetricsAggregation.new(value)}
+
+      # TODO: handle other MetricsAggregations
+      {key, value} ->
+        {key, Snap.MetricsAggregation.new(value)}
```

Comment:
ℹ️ I found this one to be a breaking change for an edge case in our end:

For an ES query like this:
```json
{
  "aggs": {
    "an_unknown_field": {
      "filter": {
        "terms": {
          "an_existing_field": ["any-value-here"]
        }
      }
    }
  },
  "from": 0,
  "query": {
    "bool": {
      "must": [
        {
          "term": {
            "id": "an-existing-id"
          }
        }
      ]
    }
  },
  "size": 0
}
```

The response is something like that:
```json
{
  "took": 1,
  "timed_out": false,
  "_shards": {
    "total": 1,
    "successful": 1,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": {
      "value": 0,
      "relation": "eq"
    },
    "max_score": null,
    "hits": []
  },
  "aggregations": {
    "an_unknown_field": {
      "doc_count": 0
    }
  }
}
```

Before this change our application was working correctly parsing result with a pattern match on `Snap.Aggregation`:
```elixir
%Snap.Aggregation{doc_count: count}
```

Now it would be broken.

The change here might still be correct and probably our application the one to be fixed, but leaving the comment here in case you want consider a backwards compatible approach or target a major release.

Reply:
I think the issue is because the struct MetricAggregation is used to map all the aggregations coming from the response while the previous Aggregation struct is used to map only the "buckets" ones. I would suggest to do the oposite, use MetricAggregation struct to map explicitly the metric aggregations and fallback into the Aggregation struct

```suggestion
      {key, %{"count" => _, "min" => _, "max" => _, "avg" => _, "sum" => _} = value} ->
        {key, Snap.MetricsAggregation.new(value)}
      {key, value} ->
        {key, Snap.Aggregation.new(value)}
```